content/concepts/discovery-routing/mDNS.md

```diff
@@ -0,0 +1,70 @@
+---
+title: "mDNS"
+description: "mDNS uses a multicast system of DNS records over a local network to enable peer discovery."
+weight: 224
+---
+
+## What is mDNS?
+
+mDNS, or multicast Domain Name System, is a way for nodes to use a multicast system
+of DNS records over a local network to discover and communicate with nodes. Nodes
+broadcast topics they're interested in instead of querying a central name server.
+The discovery, however, is limited to the peers in the local network. mDNS is commonly
+used on home networks to allow devices such as computers, printers, and smart TVs to
+find each other and connect. It uses a protocol called multicast to broadcast messages
+on the network, allowing devices to discover each other and exchange information.
+
+## mDNS in libp2p
+
+In libp2p, mDNS is used for peer discovery, allowing peers to find and
+communicate with each other on the same local network without any prior configuration.
+This is achieved through multicast DNS (mDNS) records, which are sent to all nodes on the
+local network.
+
+To initiate peer discovery, a peer sends a query to all other peers on the network using
+a DNS message with the question `_p2p._udp.local PTR`. In response, each peer will send a
+DNS message containing their discovery details. These details are stored in the additional
+records of the response and include the multiaddresses that the peer is listening on, as
+well as other information such as the peer's `<peer-name>` and `<host-name>`.
+
+The response message (that is, the answer to the DNS message) is in the form of a
+DNS record: `<service-name> PTR <peer-name>.<service-name>`. Additional record in the
+response message contains a peer's discovery details in the following form:
+`<peer-name>.<service-name> TXT "dnsaddr=..."`
+
+- `<peer-name>` is a unique case-insensitive identifier for the peer, although it is not
```

Comment:
```suggestion
- `<peer-name>` is supposed to be a unique case-insensitive identifier for the peer, required by the mDNS protocol. However, it doesn't serve any meaningful..
```

Reply:
Thinking about it, I'm not sure if we need to mention this at all. This seems to be a bit too low-level for documentation. wdyt?